simple-streaming-app/src/main/scala/com/typesafe/spark/SimpleStreamingApp.scala

```diff
@@ -58,55 +77,68 @@ object SimpleStreamingApp {
     System.exit(0)
   }
 
-  private def format(batchTime: Long, stats: (Int, (Int, Long, Double, Double, Long))): String = {
-    s"${stats._2._5}\t$batchTime\t${stats._1}\t${stats._2._1}\t${stats._2._2}\t${stats._2._3}\t${stats._2._4}"
+  private def format(batchTime: Long, input: Int, stats: Stats): String = {
+    s"${stats.millis}\t$batchTime\t$input\t${stats.count}\t${stats.sum}\t${stats.mean}\t${stats.stdDev}"
   }
 
-  /**
-   * Returns count, sum, mean and standard deviation
+  /** Returns count, sum, mean and standard deviation
    *
    */
-  private def stats(value: Iterable[Long]): (Int, Long, Double, Double, Long) = {
+  private def stats(value: Iterable[Long]): Stats = {
     val (count, sum, sqrsum) = value.foldLeft((0, 0L, 0L)) { (acc, v) =>
       // acc: count, sum, sum of squared
       (acc._1 + 1, acc._2 + v, acc._3 + v * v)
     }
     val mean = sum.toDouble / count
     val stddev = math.sqrt(count * sqrsum - sum * sum) / count
-    (count, sum, mean, stddev, System.currentTimeMillis())
+    Stats(count, sum, mean, stddev, System.currentTimeMillis())
   }
 
-  private def usageAndExit(message: String): Nothing = {
-    println(s"""$message
-Usage:
-  SimpleStreamingApp <stream_hostname> <stream_port> <congestion strategy> <batch interval milliseconds>""")
-    System.exit(1)
-    throw new Exception("Never reached")
+  val DefaultConfig = Config("local[*]", "", -1, 1000, "ignore", 1, 1)
+
+  private val parser = new OptionParser[Config]("simple-streaming") {
+    help("help")
+      .text("Prints this usage text")
```

Comment:
we need to do a doc update pass on this project after we merge this pull request.